packages/cloud_functions/cloud_functions_platform_interface/lib/src/method_channel/method_channel_https_callable_streams.dart

```diff
@@ -0,0 +1,25 @@
+// ignore_for_file: require_trailing_commas
+// Copyright 2020, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+
+import '../../cloud_functions_platform_interface.dart';
+
+class MethodChannelHttpsCallableStreams<R>
+    extends HttpsCallableStreamsPlatform<R> {
+  MethodChannelHttpsCallableStreams(String? name, String? origin, Uri? uri)
+      : _channel = EventChannel('plugins.flutter.io/firebase_functions/$name'),
```

Comment:
you have `name` as first argument but it should probably be `origin`. You are passing in origin as first arg here: https://github.com/firebase/flutterfire/pull/17214/files#diff-39179fb42585c5c75b6cfd6b3e5cd34e92705b1df98759afa825ef6ccadeb4eeR57

and here: https://github.com/firebase/flutterfire/pull/17214/files#diff-39179fb42585c5c75b6cfd6b3e5cd34e92705b1df98759afa825ef6ccadeb4eeR62

Need to do an assert here as `name` could be null. If `name` and `uri` are null, need to throw assertion error. The last segment of event channel name should be either `name` or `uri` depending on what is not `null`. Double check if there are any characters that could be a problem in event channel name as a uri might not be possible.

Reply:
The assertion is already handled on the parent(`HttpsCallableStreamsPlatform`) class: [here](https://github.com/firebase/flutterfire/pull/17214/files/24acdda0c1ea4de049c5aee7b022bc74227f98c6#diff-b44ab51c2b3d3b881d02fe73877f67674dd975acb5ff3bf0311af2c9bee86844R8)

Reply:
Wonderful, is the assertion correct? Shouldn't it be if both are null? (i.e. `&&`)